File: crypto-winter-sem-2024/mono-cipher.py
def generate_cipher_key(shift):
    alphabet = 'abcdefghijklmnopqrstuvwxyz'
    shifted_alphabet = alphabet[shift:] + alphabet[:shift]
    key = dict(zip(alphabet, shifted_alphabet))
    return key
def encrypt(message, key):
    encrypted_message = ''
    for char in message:
        if char.isalpha():
            if char.islower():
                encrypted_message += key[char]
            else:
                encrypted_message += key[char.lower()].upper()
        else:
            encrypted_message += char
    return encrypted_message
def decrypt(ciphertext, key):
    reverse_key = {v: k for k, v in key.items()}
    decrypted_message = ''
    for char in ciphertext:
        if char.isalpha():
            if char.islower():
                decrypted_message += reverse_key[char]
            else:
                decrypted_message += reverse_key[char.lower()].upper()
        else:
            decrypted_message += char
    return decrypted_message
def main():
    shift = int(input("Enter the shift value for the cipher: "))
    key = generate_cipher_key(shift)
    
    plaintext = input("Enter the message to encrypt: ")
    encrypted = encrypt(plaintext, key)
    print("Encrypted message:", encrypted)
    decrypted = decrypt(encrypted, key)
    print("Decrypted message:", decrypted)

if __name__ == "__main__":
    main()

File: crypto-winter-sem-2024/monocipher.cpp
#include <iostream>
using namespace std;

string encrypt(string text)
{
    string key = "QWERTYUIOPASDFGHJKLZXCVBNM";
    string encrypted_text = "";
    for (int i = 0; i < text.size(); i++)
    {
        encrypted_text += key[text[i] - 'A'];
    }
    return encrypted_text;
}

string decrypt(string encrypted_text)
{
    string key = "QWERTYUIOPASDFGHJKLZXCVBNM";
    string original = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string text = "";
    for (int i = 0; i < encrypted_text.size(); i++)
    {
        int index;
        for (int j = 0; j < 26; j++)
        {
            if (key[j] == encrypted_text[i])
            {
                index = j;
            }
        }
        text += original[index];
    }
    return text;
}
int main()
{
    string text;
    cout << "Enter text (IN CAPITAL): ";
    cin >> text;
    cout << "Encrypted text: " << encrypt(text) << endl;
    cout << "Decrypted text: " << decrypt(encrypt(text)) << endl;
    return 0;
}